Support rate limiting

dbast opened the pull request:
Rate limited scanning improves scanning releases with many files with a free public API token.

dbast commented:
@crazy-max Can you approve the workflow runs.. there are maybe some tests that fail with this PR to be looked into. thanks

crazy-max commented:
Thanks for your contrib! Looks like there are [multiple changes in this PR not related to each other](https://github.com/crazy-max/ghaction-virustotal/blob/master/.github/CONTRIBUTING.md#submitting-a-pull-request). Can you remove the collapsible support change and open another PR for it please?

crazy-max reviewed (changes requested):
Thanks for your contrib! Looks like there are [multiple changes in this PR not related to each other](https://github.com/crazy-max/ghaction-virustotal/blob/master/.github/CONTRIBUTING.md#submitting-a-pull-request). Can you remove the collapsible support change and open another PR for it please?

dbast commented:
@crazy-max Thanks for taking a look. Split out the collapsible change into https://github.com/crazy-max/ghaction-virustotal/pull/128 and will update+rebase this PR once it got accepted.

crazy-max commented:
> will update+rebase this PR once it got accepted.

Don't need to wait, I'm ok to review when this one has collapsible feature removed and also commits squashed.

dbast commented:
Removed the collapsible feature, rebased, updated and tested the PR.

The 10k / min as default rate_limit is an arbitrary number to simulate the previous behaviours where nothing was limited. Maybe setting it to 4 / min as default (API request limit of the free API token) would make more sense, but that is a behaviour change for those who already use a less limited payed token.

This should also close https://github.com/crazy-max/ghaction-virustotal/issues/96 

This can be enhanced in the future by making it dynamic by asking via API what the current limit is for the used token, but that is afaik much more code to write.

dbast commented:
@crazy-max Would be awesome to get another review on this. thanks!

this and the recently merged collapsible + large file support should make it possible to basically use the action for any repo with release artifacts of any kind (few to many, small or large etc.)

dbast commented:
ping

crazy-max commented on README.md:
```diff
@@ -149,6 +149,7 @@ Following inputs can be used as `step.with` keys
 | `monitor_path`**¹**         | String  | A path relative to current monitor user root folder to upload files (default `/`) |
 | `update_release_body`**²**  | Bool    | If enabled, analysis link(s) will be appended to the release body (default `false`) |
 | `github_token`**³**         | String  | [GitHub Token](https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token) used to create an authenticated client for GitHub API as provided by `secrets` (default `${{ github.token }}`) |
+| `request_rate`              | String  | API request-rate in requests/minute. Set to 4 or lower when using the standard free public API and scanning more than 4 files per run. Default: 1000 |
```
```suggestion
| `request_rate`              | String  | API request-rate in requests/minute. Set to 4 or lower when using the standard free public API and scanning more than 4 files per run (default `1000`) |
```

crazy-max commented on src/context.ts:
```diff
@@ -30,7 +31,8 @@ export async function getInputs(): Promise<Inputs> {
     updateReleaseBody: core.getBooleanInput('update_release_body'),
-    githubToken: core.getInput('github_token')
+    githubToken: core.getInput('github_token'),
+    requestRate: parseInt(core.getInput('request_rate'))
```
Define default here too, just in case the action is invoked outside a GitHub Runner:

```suggestion
    requestRate: parseInt(core.getInput('request_rate') || '1000')
```

crazy-max reviewed (changes requested):
Minor changes otherwise LGTM

> The 10k / min as default rate_limit is an arbitrary number to simulate the previous behaviour where nothing was limited. Maybe setting it to 4 / min as default (API request limit of the free API token) would make more sense, but that is a behaviour change for those who already use a less limited payed token.

Hum yeah agree limiting to `4` is a breaking change. I'm fine with `1000`.

> This can be enhanced in the future by making it dynamic by asking via API what the current limit is for the used token, but that is afaik much more code to write.

Not sure if their API returns this kind of info atm but would be nice to adjust limit automatically. Let's do that in a follow-up.

dbast commented:
thanks for the review. updated accordingly :)

crazy-max reviewed (approved):
LGTM thanks!
